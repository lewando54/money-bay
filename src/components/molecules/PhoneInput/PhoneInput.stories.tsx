/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import PhoneInput from './PhoneInput'

const meta = {
	component: PhoneInput,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof PhoneInput>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
