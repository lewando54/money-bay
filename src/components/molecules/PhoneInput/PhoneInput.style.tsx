import { StyleSheet } from 'react-native'

import globalThemeStyle, { BODY_TEXT_COLOR } from '@styling/GlobalTheme.style'
export default StyleSheet.create({
    mainContainer: {
        paddingHorizontal: 20
    },
    containerOrientation: {
        flexDirection: 'row', 
        alignItems: 'center',
        gap: 7
    },
    flag: {
        width: 20, 
        resizeMode: 'contain'
    },
    text: {
        ...globalThemeStyle.text_Regular,
        fontSize: 14, 
        color: BODY_TEXT_COLOR
    },
    textField: {
        width: 100
    }
})
