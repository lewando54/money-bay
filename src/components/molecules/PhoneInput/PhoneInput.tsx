import React from 'react'
import { Image, Platform,Text, TextInput, View } from 'react-native'

import UniversalContainer from '../../atoms/UniversalContainer/UniversalContainer'

import phoneInputStyle from './PhoneInput.style'

interface IPhoneInputProps {
    value?: string
    onChange?: (newValue: string) => void
    testID?: string
}

export default function PhoneInput({
    value,
    onChange,
    testID
}: IPhoneInputProps){
    const onChangeHandle = (newValue: string) => {
        if(onChange){
            onChange(newValue)
        }
    }

    return (
        <UniversalContainer style={[phoneInputStyle.mainContainer, phoneInputStyle.containerOrientation]} castShadow={true}>
            <View style={phoneInputStyle.containerOrientation}>
                <Image style={phoneInputStyle.flag} source={require('./assets/flag.png')} />
                <Text style={phoneInputStyle.text}>+17</Text>
            </View>
            <Text style={phoneInputStyle.text}>|</Text>
            <TextInput 
                testID={testID}
                value={value} 
                onChangeText={onChangeHandle} 
                style={[phoneInputStyle.textField, phoneInputStyle.text]} 
                placeholder='xxxxxxxxx'
                maxLength={9} 
                keyboardType={Platform.select({android: 'numeric', ios: 'number-pad'})} 
            />
        </UniversalContainer>
    )
}
