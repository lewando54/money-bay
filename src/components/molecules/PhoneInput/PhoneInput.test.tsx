import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import PhoneInput from './PhoneInput'
describe('PhoneInput', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<PhoneInput />)
    expect(tree).toMatchSnapshot()
  })
})
