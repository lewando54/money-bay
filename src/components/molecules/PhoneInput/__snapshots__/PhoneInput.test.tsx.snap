// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInput should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      [
        {
          "paddingHorizontal": 20,
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 7,
        },
      ],
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 7,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "resizeMode": "contain",
          "width": 20,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
        ]
      }
    >
      +17
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 16,
        },
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 14,
        },
      ]
    }
  >
    |
  </Text>
  <TextInput
    keyboardType="number-pad"
    maxLength={9}
    onChangeText={[Function]}
    placeholder="xxxxxxxxx"
    style={
      [
        {
          "width": 100,
        },
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 14,
        },
      ]
    }
  />
</View>
`;
